pyccel/codegen/printing/ccode.py

```diff
@@ -673,6 +707,32 @@ def init_stc_container(self, expr, assignment_var):
         init = f'{container_name} = c_init({dtype}, {keyraw});\n'
         return init
 
+    def invalidate_stc_headers(self, imports):
+        """
+        Invalidate STC headers when STC extension headers are present.
+
+        This function iterates over the list of imports and removes any targets
+        from STC headers if the target is present in their corresponding
+        STC extension headers are present.
+        The STC extension headers take care of including the standard
+        headers making them redundant.
```

Comment:
This docstring was ambiguous so I updated it. The idea is that, say i need something from the set extensions for int then simply by importing `Set_extensions` this will directly import the `set` header in STC thus there is no need to import the `set` header for int.